persist-cli/src/main/java/io/ballerina/persist/cmd/PersistCodeGeneratorTool.java

```diff
@@ -227,6 +242,19 @@ private void generateSources(String datastore, Module entityModule, String targe
         }
     }
 
+    private void generateTestSources(String testDatastore, Module entityModule, String targetModule,
+                                     Path projectPath, Path generatedSourceDirPath) {
+        if (testDatastore != null) {
+            try {
+                SourceGenerator sourceCreator = new SourceGenerator(projectPath.toString(), generatedSourceDirPath,
+                        targetModule, entityModule);
+                sourceCreator.createTestDataSources(testDatastore);
+            } catch (BalException e) {
+                errStream.printf("ERROR: %s%n", e.getMessage());
```

Comment:
Should we specifically say that only test data source creation failed here? Because it will not stop the build or conflict with the persist client generation. I mean will it be confusing if you have an error in the middle and then get the log `Persist client and entity types generated successfully in the ...` . 

Suggesting something like this as the error message,

`errStream.printf("ERROR: Test data source creation failed. %s%n", e.getMessage());`